src/uwtools/drivers/forecast.py

```diff
@@ -231,14 +231,47 @@ def _define_boundary_files(self) -> Dict[str, str]:
 
         return boundary_files
 
+    def _prepare_and_run_batch_script(self, run_directory: str) -> bool:
+        """
+        Prepares and runs batch script.
+        """
+        batch_script = self.batch_script()
+        assert self._batch_script is not None
+        outpath = Path(run_directory) / self._batch_script
+
+        if self._dry_run:
+            # Apply switch to allow user to view the run command of config.
+            # This will not run the job.
+            log.info("Batch Script:")
+            batch_script.dump(None)
+            return True
+
+        batch_script.dump(outpath)
+        return self.scheduler.submit_job(outpath)
+
     def _prepare_config_files(self, run_directory: Path) -> None:
         """
         Collect all the configuration files needed for FV3.
         """
-
-        self.create_field_table(run_directory / "field_table")
-        self.create_model_configure(run_directory / "model_configure")
-        self.create_namelist(run_directory / "input.nml")
+        if self._dry_run:
+            for call in ("field_table", "model_configure", "input.nml"):
+                log.info(f"Would prepare: {run_directory}/{call}")
+        else:
+            self.create_field_table(run_directory / "field_table")
+            self.create_model_configure(run_directory / "model_configure")
+            self.create_namelist(run_directory / "input.nml")
+
+    def _prepare_and_execute_command(self) -> bool:
+        """
+        Collects the necessary MPI environment variables in order to construct full run command.
+        """
+        pre_run = self._mpi_env_variables(" ")
+        full_cmd = f"{pre_run} {self.run_cmd()}"
+        if self._dry_run:
+            log.info("Would run: ")
+            print(full_cmd, file=sys.stdout)
```

Comment:
This should also be logged instead of `print()`ed in dry-run mode. And no need for the space after the colon in the logged string in the preceding command: There will be a newline there anyway.